src/credentials-manager/index.js

```diff
@@ -0,0 +1,147 @@
+import {Platform, NativeModules} from 'react-native';
+import CredentialsManagerError from './credentialsManagerError';
+
+export default class CredentialsManager {
+  constructor(auth) {
+    const {clientId, domain} = auth;
+    this.domain = domain;
+    this.clientId = clientId;
+    this.Auth0Module = NativeModules.A0Auth0;
+  }
```

Comment:
Would it not be simpler to just accept discreet parameters here?

```js
constructor(domain, clientId) {
  // ..
}
```

Reply:
If we're going to keep it as an object, the constructor should have doc comments to document what members to put on the object.

Reply:
I did this to keep it consistent with [how we initialize WebAuth currently](https://github.com/auth0/react-native-auth0/blob/master/index.js#L24).

Credentials manager should exposed through the Auth0 object like this

```
const auth0 = new Auth0({
  domain: '{YOUR_AUTH0_DOMAIN}',
  clientId: '{YOUR_CLIENT_ID}',
});

await auth0.credentialsManager.getCredentials()
```

This would mean that the user doesn't need to construct this object directly.

Reply:
I would still suggest the doc comments would be useful, for ourselves and our contributors. But I can see the argument for keeping it as an object 👍🏻